wagtail/images/models.py

```diff
@@ -1384,16 +1384,35 @@ def get_upload_to(self, filename):
     def check(cls, **kwargs):
         errors = super().check(**kwargs)
         if not cls._meta.abstract:
-            if not any(
+            in_unique_together = any(
                 set(constraint) == {"image", "filter_spec", "focal_point_key"}
                 for constraint in cls._meta.unique_together
-            ):
+            )
+
+            # We only include fields to check if they are a UniqueConstraint, and
+            # they have no conditions on them.
+            unique_constraint_fields = [
+                constraint.fields
+                for constraint in cls._meta.constraints
+                if (
+                    isinstance(constraint, UniqueConstraint)
+                    and constraint.condition is None
+                )
+            ]
+            in_unique_constraint = any(
+                set(constraint) == {"image", "filter_spec", "focal_point_key"}
+                for constraint in unique_constraint_fields
+            )
+
+            # We check for some kind of unique constraint, whether in unique together or
+            # as part of the constraints.
+            if not in_unique_together and not in_unique_constraint:
                 errors.append(
                     checks.Error(
                         "Custom rendition model %r has an invalid unique_together setting"
                         % cls,
-                        hint="Custom rendition models must include the constraint "
-                        "('image', 'filter_spec', 'focal_point_key') in their unique_together definition.",
+                        hint="Custom rendition models must include a unique constraint "
+                        "on the 'image', 'filter_spec', and 'focal_point_key' fields.",
```

Comment:
I think this should be the new error message itself (right now the error message still uses `unique_together`), and the `hint` should be something the developer can use, e.g. suggesting

```py
                        "Add models.UniqueConstraint(fields={'image', 'filter_spec', 'focal_point_key'}, "
                        "name='unique_rendition') to %s.Meta.constraints."
                        % (cls.__name__,)
```